app/components/channel_drawer/channels_list/channel_item/channel_item.js

```diff
@@ -0,0 +1,129 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React, {PureComponent} from 'react';
+import PropTypes from 'prop-types';
+import {
+    TouchableHighlight,
+    Text,
+    View
+} from 'react-native';
+
+import ChannelIcon from 'app/components/channel_icon';
+import {wrapWithPreventDoubleTap} from 'app/utils/tap';
+import {changeOpacity, makeStyleSheetFromTheme} from 'app/utils/theme';
+
+export default class ChannelItem extends PureComponent {
+    static propTypes = {
+        channelId: PropTypes.string.isRequired,
+        currentChannelId: PropTypes.string.isRequired,
+        displayName: PropTypes.string.isRequired,
+        onSelectChannel: PropTypes.func.isRequired,
+        status: PropTypes.string,
+        type: PropTypes.string.isRequired,
+        theme: PropTypes.object.isRequired
+    };
+
+    onPress = wrapWithPreventDoubleTap(() => {
+        const {channelId, currentChannelId, displayName, onSelectChannel} = this.props;
+        requestAnimationFrame(() => {
+            onSelectChannel({id: channelId, display_name: displayName}, currentChannelId);
+        });
+    });
+
+    render() {
+        const {
+            channelId,
+            currentChannelId,
+            displayName,
+            status,
+            theme,
+            type
+        } = this.props;
+
+        const style = getStyleSheet(theme);
+        const isActive = channelId === currentChannelId;
+
+        let activeItem;
+        let activeText;
+        let activeBorder;
+
+        if (isActive) {
+            activeItem = style.itemActive;
+            activeText = style.textActive;
+
+            activeBorder = (
+                <View style={style.borderActive}/>
+            );
+        }
+
+        const icon = (
+            <ChannelIcon
+                isActive={isActive}
+                hasUnread={false}
+                membersCount={displayName.split(',').length}
```

Comment:
This will have issues if someone includes a comma in their name. That's probably fine since it'll be very rare, and it won't have any significant side effects, but is there a nicer way to do this like having the ChannelIcon connect to the store if the type is a group message?

Reply:
I'll see what I can do with this one, good idea

Reply:
I'm not going to change this as we don't have the channel members at this point, we'll need to do something here, but I think that should be another ticket